internals/secrethub/service_init_test.go

```diff
@@ -0,0 +1,244 @@
+package secrethub
+
+import (
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/secrethub/secrethub-cli/internals/cli/filemode"
+	"github.com/secrethub/secrethub-cli/internals/cli/ui/fakeui"
+	"github.com/secrethub/secrethub-go/internals/api"
+	"github.com/secrethub/secrethub-go/internals/assert"
+	"github.com/secrethub/secrethub-go/internals/errio"
+	"github.com/secrethub/secrethub-go/pkg/secrethub"
+	"github.com/secrethub/secrethub-go/pkg/secrethub/credentials"
+	"github.com/secrethub/secrethub-go/pkg/secrethub/fakeclient"
+)
+
+func TestServiceInitCommand_Run(t *testing.T) {
+	keyCreator := credentials.CreateKey()
+	_ = keyCreator.Create()
+	exportedCredential, _ := keyCreator.Export()
+	testErr := errio.Namespace("test").Code("test").Error("test error")
+
+	cases := map[string]struct {
+		cmd             ServiceInitCommand
+		serviceService  fakeclient.ServiceService
+		setFunc         func(path string, permission string, accountName string) (*api.AccessRule, error)
+		expectedPerm    *api.AccessRule
+		newClientErr    error
+		expectedFileOut []byte
+		expectedOut     string
+		fileErr         error
+		err             error
+	}{
+		"success": {
+			cmd: ServiceInitCommand{
+				repo:       api.RepoPath("test/repo"),
+				credential: keyCreator,
+			},
+			serviceService: fakeclient.ServiceService{
+				CreateFunc: func(path string, description string, credentialCreator credentials.Creator) (*api.Service, error) {
+					return &api.Service{}, nil
+				},
+			},
+			expectedOut: string(exportedCredential) + "\n",
+		},
+		"write to file": {
+			cmd: ServiceInitCommand{
+				repo:       api.RepoPath("test/repo"),
+				credential: keyCreator,
+				file:       "test.txt",
+				fileMode:   filemode.New(os.ModePerm),
+			},
+			serviceService: fakeclient.ServiceService{
+				CreateFunc: func(path string, description string, credentialCreator credentials.Creator) (*api.Service, error) {
+					return &api.Service{
+						ServiceID: "testService",
+					}, nil
+				},
+			},
+			expectedFileOut: []byte(string(exportedCredential) + "\n"),
+			expectedOut:     "Written account configuration for testService to test.txt. Be sure to remove it when you're done.\n",
+		},
+		"fail write to file": {
+			cmd: ServiceInitCommand{
+				repo:       api.RepoPath("test/repo"),
+				credential: keyCreator,
+			},
+			serviceService: fakeclient.ServiceService{
+				CreateFunc: func(path string, description string, credentialCreator credentials.Creator) (*api.Service, error) {
+					return &api.Service{
+						ServiceID: "testService",
+					}, nil
+				},
+			},
+			err: ErrCannotWrite("path/test.txt", "open path/test.txt: no such file or directory"),
```

Comment:
We're not actually setting a `fileErr` in this test case. Should we?

Reply:
Yes, we should. Fixed the test.